ext/SciMLBaseChainRulesCoreExt.jl

```diff
@@ -114,4 +114,15 @@ function ChainRulesCore.rrule(::SciMLBase.EnsembleSolution, sim, time, converged
     out, EnsembleSolution_adjoint
 end
 
+function ChainRulesCore.rrule(
+        ::Type{<:SciMLBase.NonlinearSolution{
+            T, N, uType, R, P, A, O, uType2, S, Tr}}, u, resid, prob,
+        args...) where {T, N, uType, R, P, A, O, uType2, S, Tr}
+    function NonlinearSolutionAdjoint(ȳ)
+        (NoTangent(), ȳ.u, NoTangent(), ŷ.prob, ntuple(_ -> NoTangent(), length(args))...)
```

Comment:
As a follow up, is the type of  `ȳ` a solution type still with the latest SciMLSensitivity